### PR TITLE
[Issue Tracker] Add upgrade script for module column

### LINKED ISSUE #5959: [Issue Tracker] Write upgrade script for module column
The issue tracker has a column called "module". This was an integer which was referencing the LorisMenu table (without a foreign key). The table was dropped and the column became a foreign key to the modules table, which makes more sense for a column called "module"..

We need to:
1. Declare the new foreign key properly, now that the column is referencing something that makes sense.
2. Write a script which fixes the data before the release patch drops the LorisMenu table. This is tricky, since the issue tracker was previously calling something that has no relation to modules a "module" and we need a way to map the data. Options I can think of are:
 a. Null the data so that the "module" column for existing issues don't change randomly 
 b. Try and map the module getLongName() function to the previous LorisMenu label and warn/null the ones that can't be mapped.
 c. Try and parse the URL from the menu item to get the module name and null the ones that can't be mapped.

This needs to be done before the next release.

### PR DESCRIPTION
**Pull request by johnsaigle**
## Brief summary of changes

Issue Tracker has a `module` table which references IDs in the deleted LorisMenu table.

This PR adds a script to map the old LorisMenu IDs in this column to the corresponding IDs in the new `modules` table.

#### Testing instructions (if applicable)

1. Your database must have both the `LorisMenu` table and the `modules` table.
- If you've already installed `modules` and dropped `LorisMenu`, download and run [this SQL file](https://raw.githubusercontent.com/aces/Loris/2f07cc6d75a7f78201748ec917e963062972fa8a/SQL/0000-00-02-Menus.sql)
- If you still have `LorisMenu` and do not have `modules`, download and run [this SQL file](https://raw.githubusercontent.com/driusan/Loris/ec0981150a29bed86eece3b3a3452d9f166cf72a/SQL/0000-00-02-Modules.sql).

2. Run my script. It should display a before and after comparison of the values as well as the SQL statements you need to run, e.g. on Raisinbread:

```SQL
Issue ID: 16. Module before (`LorisMenu`): `dataquery/`. Module after (`modules`): `dataquery`
Issue ID: 19. Module before (`LorisMenu`): `examiner/`. Module after (`modules`): `examiner`
Issue ID: 20. Module before (`LorisMenu`): `media/`. Module after (`modules`): `media`
Issue ID: 26. Module before (`LorisMenu`): `examiner/`. Module after (`modules`): `examiner`
Issue ID: 27. Module before (`LorisMenu`): `conflict_resolver/`. Module after (`modules`): `conflict_resolver`
UPDATE `issues` SET module=14 WHERE issueID=16;
UPDATE `issues` SET module=17 WHERE issueID=19;
UPDATE `issues` SET module=29 WHERE issueID=20;
UPDATE `issues` SET module=17 WHERE issueID=26;
UPDATE `issues` SET module=9 WHERE issueID=27;
ALTER TABLE `issues` ADD FOREIGN KEY (moduleID) REFERENCES (modules(ID));
```

#### Link(s) to related issue(s)

* Resolves #5959

### CODE REVIEW

File: tools/single_use/22.0_To_23.0.php [hunk 1]
```diff
@@ -0,0 +1,133 @@
+<?php declare(strict_types=1);
+/**
+ * This script is used to correctly establish a link betwen the `issues` table
+ * used by the Issue Tracker module and the new table `modules`.
+ * It outputs SQL statements necessary to upgrade this table from LORIS version
+ * 22 to version 23.
+ * It relies on the existence of the `LorisMenu` table and the `modules` table
+ * so it must be used in the following way:
+ *
+ * 1. A user must apply the patch file:
```
Comment (johnsaigle) [inline tools/single_use/22.0_To_23.0.php]: @ridz1208 @driusan Please let me know if this is the upgrade process you had in mind. I'm happy to modify the script to make the upgrade process smooth.

**Comment by johnsaigle** [comment]
@ridz1208 What's the best way to apply these changes to Raisinbread?

**Comment by ridz1208** [comment]
source RB, run your tool, dump RB

**Comment by johnsaigle** [comment]
I'm not sure where the foreign key should be added since `0000-00-00-schema.sql` creates `issues` but it needs a FK references to `0000-00-02-Modules.sql`. Any preference?

**Comment by driusan** [comment]
This shouldn't be in the changelog, it should be in the release notes. This is a one time use script that no one cares about after their upgrade is finished.

File: tools/single_use/22.0_To_23.0.php (hunk 1 above)
Comment (driusan) [reply tools/single_use/22.0_To_23.0.php]: We'll need to change the patch name before the release and make sure it's not duplicated in the release patch, but yes, this seems reasonable.

Comment (ridz1208) [reply tools/single_use/22.0_To_23.0.php]: adressed in my PR

**Comment by ridz1208** [comment]
added blocked and needs changelog update.

blocked by https://github.com/aces/Loris/pull/6120 which contains a notes for existing projects section where this can be disclosed

File: SQL/0000-00-02-Modules.sql [hunk 2]
```diff
@@ -47,3 +47,5 @@ INSERT INTO modules (Name, Active) VALUES ('survey_accounts', 'Y');
 INSERT INTO modules (Name, Active) VALUES ('timepoint_list', 'Y');
 INSERT INTO modules (Name, Active) VALUES ('user_accounts', 'Y');
 INSERT INTO modules (Name, Active) VALUES ('electrophysiology_browser', 'Y');
+
+ALTER TABLE issues ADD CONSTRAINT `fk_issues_7` FOREIGN KEY (`module`) REFERENCES `modules` (`ID`);
```
Comment (driusan) [inline SQL/0000-00-02-Modules.sql]: This doesn't belong in the Modules.sql, it belongs in wherever the 'issues' table is defined in the default schema.

File: tools/single_use/Convert_LorisMenuID_to_ModuleID.php [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+unset($result);
+
+
+// Use mapping arrays to replace old `LorisMenu` ID values in issue_tracker
+// `module` column with new `modules` table ID values.
+$result = $DB->pselect('SELECT issueID,module FROM issues', array());
+$issueTrackerMapping = [];
+foreach ($result as $row) {
+    // Skip rows in this table that do not have a module associated with them.
+    if (is_null($row['module'])) {
```
Comment (driusan) [inline tools/single_use/Convert_LorisMenuID_to_ModuleID.php]: just add `WHERE module IS NOT NULL` to the select statement.

File: SQL/0000-00-02-Modules.sql (hunk 2 above)
Comment (ridz1208) [reply SQL/0000-00-02-Modules.sql]: cant add a foreign key to a table that doesnt exist

Comment (driusan) [reply SQL/0000-00-02-Modules.sql]: I see.. I still don't like it, but you have a point.

Comment (ridz1208) [reply SQL/0000-00-02-Modules.sql]: same here... it's the consequence of having multiple schema files

**Pull request merged**